test/e2e/density.go

```diff
@@ -0,0 +1,191 @@
+/*
+Copyright 2015 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+// Delete a Replication Controller and all pods it spawned
+func DeleteRC(c *client.Client, rc *api.ReplicationController, ns, name string) error {
+	rc.Spec.Replicas = 0
+	if _, err := c.ReplicationControllers(ns).Update(rc); err != nil {
+		return fmt.Errorf("Failed to resize replication controller %s to zero: %v", name, err)
+	}
```

Comment:
Proper client behavior is to wait (poll or watch) until rc.Status.Replicas also equals zero. Otherwise, you'll leak pods if controller manager is temporarily down.